### PR TITLE
bin,lib: remove extra comments/lines/spaces

### PR DESCRIPTION
**Pull request by maclover7**
##### Checklist

- [x] `npm install && npm test` passes
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Description of change

- Removes "module dependencies" comments and things that, IMHO, don't add
too much value. Happy to add back if helps some people when reading
through `node-gyp`.
- DRY up `lib/process-release.js`.
- Removes a bunch of extra blank lines, as well as random spaces.

### CODE REVIEW

**Comment by maclover7** [comment]
CI: https://ci.nodejs.org/job/nodegyp-test-pull-request/61/

File: bin/node-gyp.js [hunk 1]
```diff
@@ -126,7 +118,6 @@ process.on('uncaughtException', function (err) {
 })
 
 function errorMessage () {
-  // copied from npm's lib/util/error-handler.js
```
Comment (richardlau) [inline bin/node-gyp.js]: This is sort of an attribution (although the current path in npm now appears to be lib/utils/error-handler.js), do maybe keep?

**Comment by maclover7** [comment]
updated @richardlau, PTAL

**Comment by maclover7** [comment]
CI: https://ci.nodejs.org/job/nodegyp-test-pull-request/71/

**Comment by maclover7** [comment]
Landed in 6b0f4ae, thank you for the review!

**Pull request merged**